File: output/ChatGPT/avatar/Python/Java/atcoder_AGC038_A.java
public class atcoder_AGC038_A {
    public static void solve(int H, int W, int A, int B) {
        char[][] answer = new char[H][W];
        for (int i = 0; i < H; i++) {
            for (int j = 0; j < W; j++) {
                if (i < B) {
                    if (j < A) {
                        answer[i][j] = '0';
                    } else {
                        answer[i][j] = '1';
                    }
                } else {
                    if (j < A) {
                        answer[i][j] = '1';
                    } else {
                        answer[i][j] = '0';
                    }
                }
            }
        }
        for (int i = 0; i < H; i++) {
            System.out.println(new String(answer[i]));
        }
    }

    public static void main(String[] args) {
        solve(3, 4, 2, 1);
    }
}

File: output/ChatGPT/avatar/Python/C++/atcoder_AGC038_A.cpp
#include <iostream>
#include <vector>
using namespace std;

void solve(int H, int W, int A, int B) {
    vector<vector<char>> answer(H, vector<char>(W, '0'));
    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            if (i < B) {
                if (j >= A) {
                    answer[i][j] = '1';
                }
            } else {
                if (j < A) {
                    answer[i][j] = '1';
                }
            }
        }
    }
    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            cout << answer[i][j];
        }
        cout << endl;
    }
}

int main() {
    int H, W, A, B;
    cin >> H >> W >> A >> B;
    solve(H, W, A, B);
    return 0;
}